certbot-dns-cloudflare/certbot_dns_cloudflare/__init__.py

```diff
@@ -1 +1,76 @@
-"""Cloudflare DNS Authenticator"""
+"""
+The `~certbot_dns_cloudflare.dns_cloudflare` plugin automates the process of
+completing a ``dns-01`` challenge (`~acme.challenges.DNS01`) using the
+Cloudflare API.
+
+
+Named Arguments
+---------------
+
+========================================  =====================================
+``--dns-cloudflare-credentials``          Cloudflare credentials_ INI file.
+                                          (Required)
+``--dns-cloudflare-propagation-seconds``  The number of seconds to wait for DNS
+                                          to propagate before asking the ACME
+                                          server to verify the DNS record.
+                                          (Default: 10)
+========================================  =====================================
+
+Credentials
+-----------
+
+Use of this plugin requires a configuration file containing Cloudflare API
+credentials, obtained from your Cloudflare
+`account page <https://www.cloudflare.com/a/account/my-account>`_.
+
+.. code-block:: ini
+   :name: credentials.ini
+   :caption: Example credentials file:
+
+   # Cloudflare API credentials used by Certbot
+   dns_cloudflare_email = cloudflare@example.com
+   dns_cloudflare_api_key = 0123456789abcdef0123456789abcdef01234567
+
+The path to this file can be provided interactively or using the
+``--dns-cloudflare-credentials`` command-line argument. Certbot records the path
+to this file for use during renewal, but does not store the file's contents.
+
+.. caution::
+   You should protect these API credentials as you would the password to your
+   Cloudflare account. Users who can read this file can use these credentials
+   to issue API calls on your behalf. Users who can cause Certbot to run using
+   these credentials can complete a ``dns-01`` challenge to acquire new
+   certificates or revoke existing certificates for associated domains.
+
```

Comment:
>I agree a dedicated, locked down machine is probably the best option, but I'm not sure if that's what we should recommend to the average user (although perhaps we could mention in an offhand comment that's a way to provide additional security here).

Agreed that this isn't a good recommendation for the average user.

I think something like this probably belongs somewhere outside of the main documentation for these plugins.
 
> Minimal permissions on file containing API credentials (e.g. 600).

Certbot will warn about more generous permissions and I'll add a note about this to the documentation.

> Minimal permissions with API credentials themselves. In the rare cases it is possible, we should instruct the user to create the API credentials with the fewest permissions Certbot needs to operate.

The documentation for the Google plugin (just pushed) covers this. I'll do the same for AWS and any other providers where I can find a way to create API credentials with limited permissions.